### DISCUSSION TITLE
Help before purchase

### DISCUSSION THREAD

**Discussion by Tilgoron** [Anime4000/RTL960x#124] [General]
Hey guys :)

At home I've got a symmetrical 1Gbit Fiber connection and my ISP provided me this router: Huawei OptiXstar EG8147X6 ([Huawei](https://e.huawei.com/es/products/enterprise-transmission-access/terminal/ont/optixstar-eg8147x6))

My idea was to replace the ISP router with the Archer AX206 [Link](https://www.tp-link.com/us/home-networking/wifi-router/archer-ax206/#overview). However I wanted to avoid buying and power up a media converted, so I thought about purchasing one of those GPON/SFP adapters.

I am totally newbie about SFP modules, can please someone point me out what I should buy and/or firmware to flash?

Also, the Archer router I see it has a 10Gbit SFP+ port, will a normal GPON/SFP module work?

Here below the specs of the GPON port of my Huawei router:

• Port type: SC/APC
• Class B+
• Receiver sensitivity: –27 dBm to –29 dBm
• Overload optical power: –8 dBm
• Wavelengths: US 1310 nm, DS 1490 nm
• Wavelength Blocking Filter (WBF) of G.984.5
• Flexible mapping between the GEM port and TCONT
• GPON: consistent with the SN or password authentication defined in G.984.3
• Bi-directional FEC
• SR-DBA and NSR-DBA

**Comment by Anime4000** [comment]
it should works by emulating Huawei, also ODI firmware has 2 option, can test both

ODI Stick might work on TP-Link, if TP-Link SFP+ support 2.5G mode, you can run with 2.5G mode with unlocked 2.4Gbps Fiber download speed.

at least need SFP Media Converter if, TP-Link didn't work

**Comment by Tilgoron** [comment]
I see, so you suggest purchasing an ODI DFP-34X-2C2 | RTL9601D right? 

By looking at Aliexpress I see they have 3 different models Gpon/epon/xpon, if I understand correctly from your Readme file you suggest purchasing the Xpon with WebUI right?

Indeed, worse case scenario the TP-Link have a 2.5Gbit Ethernet WAN port, so in case it doesn't work I could just purchase a fiber media converter :)

**Reply by Anime4000** [reply]
Correct, ODI DFP-34X-2C2 | RTL9601D | XPON
Always choose XPON no matter what, that give you RTL9601D SoC like everyone else

Also check VLAN, some user need VLAN patch to work

Join this discord https://discord.gg/KUHwDgkRtR (managed by `stich86_0046#3321`)
See if anyone have same ISP as you do.

Media Converter is cheap,

My suggestion is, seperate Router and Wi-Fi, since you are going to use ONU Stick, it would be better off use enthusiast/business router like Mikrotik RB5009 (which is support 2.5G at SFP+)

Using commercial home router like TP-Link, it wont give you advanced networking and routing, troubleshooting PON Stick might require to change WAN,

With Mikrotik or Ubiquiti, you can set SFP interface an IP Address, add into NAT group together with WAN, so your LAN subnet can access PON Stick WebUI (192.168.1.1/24)

Even better doing routes, TP-Link wont let you add IP Address at SFP port since that port is WAN ingress.

**Reply by Tilgoron** [reply]
Very clear, thank you very much for your help! I will start purchasing the module first and then start testing :)

**Comment by ccaillet00** [comment]
@Vinko90 Hi there 
I want to ask you how your testing went? Are you using the ODI DFP-34X-2C2 | RTL9601D | XPON at the moment? Did you have any problems to set it up? I have the same Huawei router from my ISP provider and i want to get rid of it. 

Thank you for your answer! :)

**Reply by Tilgoron** [reply]
Hii, yes I had the same and the setup went super smooth as far as I remember. (I am not using it anymore because I am in a different house with different ISP).

The only thing I had to do was the basic configuration where you need to type the values below:

| Information | Flash Set Value | Example |
|-------------|-----------------|---------|
| MAC Address | `ELAN_MAC_ADDR` <sub><sup>*EPON user only, [V2801F](#v2801f), [DFP-34X-2C2](#dfp-34x-2c2)</sup></sub> | `781735000000` |
| Hardware Version | `HW_HWVER` <sub><sup>*[V2801F](#v2801f)</sup></sub> | `BF9.A` | 
| Software Version | `OMCI_SW_VER1`, `OMCI_SW_VER2`, `CUSTOM_OMCI_SW_VER1`, `CUSTOM_OMCI_SW_VER2` | `V3R017C10S100` |
| Serial Number | `GPON_SN` | `HWTC35000000` |
| Manufacture Info | `PON_VENDOR_ID` | `HWTC` |
| Device Model | `GPON_ONU_MODEL` | `HG8240H` |
| OUI | `OUI` | `875773` |
| Device SN | `HW_SERIAL_NO` | `UONHUWH12341234123` |

With those data I already had a connection with the ISP. 

Then it comes the part about obtaining an IP address from the ISP:
Here in Spain, with the ISP I was using the XPON, the original router was getting a DHCP address, however they do use VLAN's. That ISP specifically, was using VLAN-22 for Internet and VLAN-23 for IPTV. That being said, I had to set the WAN port of my Ubiquiti router to fetch IP as DHCP and tag with VLAN-22, that was it. If your ISP provider is using VLAN or not, you can easily get that information from the Huawei router, I don't remember where it is but there is a page where you can see the tagging.

PS: In my current house, the ISP I have also provide the same OptiXstar EG8147X6 router. The problem is that they don't allow you to log into the router dashboard. A few months ago I was in a call with customer support asking to have at least temporary access, they granted me a remote password for a day and I was able to get the VLAN ID (Without that you won't be able to get an IP address with your new router). With that, I was able to use the XPON for 3 months straight without any issues. Then one night, internet went down for a day, after some inspections I realized they PROBABLY blacklisted the XPON somehow. I had to use the original ISP router again and I realized the public IP address changed too. That means they either figure out I was not using their router, or they made some changes to the network. Now I can't login in the ISP router anymore if I don't call them, so I am back at using their router in front of my Ubiquiti. 
I am telling you this because I don't understand yet if ISP's are able to figure out from the PN if the fiber converter is theirs or not.

**Reply by Anime4000** [reply]
this likely Proprietary OMCI or TR069 is absent, this is how OLT can detect
